src/OldFileVersionCompatibilityTests/TestAssertionException.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Diagnostics;
+
+namespace OldFileVersionCompatibilityTests
+{
+    class TestAssertionException : Exception
+    {
+        public string FileName { get; }
+        public int LineNumber { get; }
+
+        public TestAssertionException(string message)
+            : base(message)
+        {
+            var trace = new StackTrace(true);
+            var frame = trace.GetFrame(2);
```

Comment:
This parameter of 2 seems a bit sketchy, since it assumes knowledge about the caller. Any chance we could make it a parameter of the ctor?

Reply:
I think that's a good point. But I argue that the class is part of a closed system where it has a prescribed purpose where the exception is intended always to be created from an assert function. And so far, all the assert functions are only one level deep. And I have no expectation they will go deeper.